File: solutions/python/guidos-gorgeous-lasagna/3/lasagna.py
"""Functions used in preparing Guido's gorgeous lasagna.

Learn about Guido, the creator of the Python language:
https://en.wikipedia.org/wiki/Guido_van_Rossum

This is a module docstring, used to describe the functionality
of a module and its functions and/or classes.
"""

EXPECTED_BAKE_TIME = 40
PREPARATION_TIME = 2


def bake_time_remaining(elapsed_time):
    """Calculate the bake time remaining.

    :param elapsed_bake_time: int - baking time already elapsed.
    :return: int - remaining bake time (in minutes) derived from 'EXPECTED_BAKE_TIME'.

    Function that takes the actual minutes the lasagna has been in the oven as
    an argument and returns how many minutes the lasagna still needs to bake
    based on the `EXPECTED_BAKE_TIME`.
    """
    return EXPECTED_BAKE_TIME - elapsed_time


# You might also consider defining a 'PREPARATION_TIME' constant.
# You can do that on the line below the 'EXPECTED_BAKE_TIME' constant.
# This will make it easier to do calculations.
def preparation_time_in_minutes(number_of_layers):
    """calculate the preparation time remaining.

    :param number_of_layers: number of layers in the lasagna.
    :return: time need to bake each of those layers
    """
    return number_of_layers * PREPARATION_TIME


def elapsed_time_in_minutes(number_of_layers, elapsed_time):
    """

    :param number_of_layers: number of layers in the lasagna.
    :param elapsed_time: time already spent baking
    :return: all the time elapsed
    """
    return preparation_time_in_minutes(number_of_layers) + elapsed_time

File: solutions/java/lasagna/2/src/main/java/Lasagna.java
public class Lasagna {
    public int expectedMinutesInOven(){
        return 40;
    }
    public int remainingMinutesInOven(int timeSpent){
        Lasagna lasagna = new Lasagna();
        return lasagna.expectedMinutesInOven() - timeSpent;
    }
    public int preparationTimeInMinutes(int layers){
        return layers * 2;
    }
    public int totalTimeInMinutes(int layers, int timeSpent){
        Lasagna lasagna = new Lasagna();
        return lasagna.preparationTimeInMinutes(layers) + timeSpent;
    }
}
